samples/Mvc.Server/Controllers/AuthorizationController.cs

```diff
@@ -0,0 +1,183 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/openiddict/openiddict-core for more information concerning
+ * the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using AspNet.Security.OpenIdConnect.Extensions;
+using AspNet.Security.OpenIdConnect.Server;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Mvc.Server.Models;
+using Mvc.Server.ViewModels.Authorization;
+using OpenIddict;
+
+namespace Mvc.Server {
+    public class AuthorizationController : Controller {
+        private readonly OpenIddictApplicationManager<OpenIddictApplication<Guid>> _applicationManager;
+        private readonly SignInManager<ApplicationUser> _signInManager;
+        private readonly OpenIddictUserManager<ApplicationUser> _userManager;
+
+        public AuthorizationController(
+            OpenIddictApplicationManager<OpenIddictApplication<Guid>> applicationManager,
+            SignInManager<ApplicationUser> signInManager,
+            OpenIddictUserManager<ApplicationUser> userManager) {
+            _applicationManager = applicationManager;
+            _signInManager = signInManager;
+            _userManager = userManager;
+        }
+
+        [Authorize, HttpGet, HttpPost, Route("~/connect/authorize")]
+        public async Task<IActionResult> Authorize() {
+            // Note: when a fatal error occurs during the request processing, an OIDC response
```

Comment:
@ilmax now that developers are responsible of creating their own controller, I believe we should handle "errored requests" directly in OpenIddict as there's a risk they won't handle them properly in their code.

I have 3 approaches in mind:
  - Display a native error page, embedded in OpenIddict.Core. This simple approach has a downside: it's not very flexible and doesn't allow customizing it.

  - Throw an exception and let the developer handle it using the error page middleware. I'm not a huge fan of throwing exceptions in this case, but it's simple and flexible.

  - Rewrite the request path to something like /connect/error and invoke the rest of the pipeline. This allows creating a corresponding "error" action/controller to display the error code.

Any thoughts?